### DISCUSSION TITLE
Arduino extension for VS Code deprecated

### DISCUSSION THREAD

**Discussion by CsFreiber** [microchip-pic-avr-solutions/avr-iot-cellular-arduino-library#28] [Q&A]
Hello community,

Microsoft has announced that the Arduino extension for VS Code is no longer supported (see: https://marketplace.visualstudio.com/items?itemName=vsciot-vscode.vscode-arduino)
![image](https://github.com/user-attachments/assets/02744755-c865-44f7-9ff9-54aa62e74c84)

As the Arduino IDE version 1. is not very user friendly, but is necessary for the DxCore library to work, are there any suggestions on how to still use VS Code? Is there a quick possibility to migrate the Arduino Code to the Microchip Studio (as it supports the AVR128DB48 Chip) and use the DxCore and avr-iot-cellular library or is there a better way?
Thank you for your help!

**Comment by simengangstad** [comment]
Hello @CsFreiber,

This is my personal account (we've spoken earlier through my former work account: @M65649). I no longer work for Microchip so I don't know what the plans are for this, but have you checked out [Arduino IDE 2](https://www.arduino.cc/en/software)? It's what I've been using lately and it works well with the DB and the cellular library. It's somewhat better than Arduino IDE 1, but not as good as just using VS Code. 

There is not a good way to migrate things to Microchip Studio or MPLAB X. I recommend just using Arduino IDE 2. If you are not afraid to go to the command line you can use VS Code + arduino-cli. I mainly used that to develop the library. There are some scripts in the repository for how to [compile](https://github.com/microchip-pic-avr-solutions/avr-iot-cellular-arduino-library/blob/main/scripts/make.sh) and [flash](https://github.com/microchip-pic-avr-solutions/avr-iot-cellular-arduino-library/blob/main/scripts/flash.sh) with arduino-cli and pymcuprog.

**Reply by CsFreiber** [reply]
Hello Simen,

we found a good way to encounter this problem by using the **platform IO** extension for VS Code ([https://platformio.org/install/ide?install=vscode](url) 
With the platform IO you can compile your code to a .hex file and flash the AVR-IoT-Cellular mini board per drag and drop.

Best regards
CsFreiber
